core/server/suspend_test.go

```diff
@@ -137,58 +139,86 @@ func TestSuspend_Suspend(t *testing.T) {
 	})
 }
 
-func checkSpec(t *testing.T, k client.Client, name types.NamespacedName, obj client.Object) bool {
+func checkSpec(t *testing.T, k client.Client, principalID string, name types.NamespacedName, obj client.Object, suspend bool) bool {
 	switch v := obj.(type) {
 	case *sourcev1.GitRepository:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 	case *kustomizev1.Kustomization:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 
 	case *helmv2.HelmRelease:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 
 	case *sourcev1b2.Bucket:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 
 	case *sourcev1b2.HelmRepository:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 
 	case *reflectorv1.ImageRepository:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 
 	case *imgautomationv1.ImageUpdateAutomation:
 		if err := k.Get(context.Background(), name, v); err != nil {
 			t.Error(err)
 		}
-
+		checkSuspendAnnotations(t, principalID, v.ObjectMeta.Annotations, name, suspend)
 		return v.Spec.Suspend
 	}
 
 	t.Errorf("unsupported object %T", obj)
 
 	return false
 }
+
+// checkSuspendAnnotations checks for the existance of suspend annotations
+// passes if suspended and annotations exist, or not suspended and annotations don't exist
+// if annotations exist, the principal is checked in the annotation for suspended-by
+func checkSuspendAnnotations(t *testing.T, principalID string, annotations map[string]string, name types.NamespacedName, suspend bool) {
+	if suspend {
+		// suspended and annotations exist check
+		if suspendedBy, ok := annotations["weave.works/suspended-by"]; ok {
+			// principal check if suspended and annotations exist
+			if suspendedBy != principalID {
+				t.Errorf("expected annotation weave.works/suspended-by to be set to the principal %s", principalID)
+			}
+		} else {
+			t.Errorf("expected annotation weave.works/suspended-by not found for %s", name.String())
```

Comment:
You don't need to call `.String()` because...
https://pkg.go.dev/fmt

> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
> 5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

See https://go.dev/play/p/G1RCTkzZDoD